File: java/Kucing.java
public class Kucing extends Hewan {
    private String ras;
    private String warnaBulu;

    public Kucing() {
        super();
    }

    public Kucing(String famili, String habitat, String jenisMakanan, String caraBerkembangbiak, int harapanHidup, Makanan food, String ras, String warnaBulu) {
        super(famili, habitat, jenisMakanan, caraBerkembangbiak, harapanHidup, food);
        this.ras = ras;
        this.warnaBulu = warnaBulu;
    }

    // Setter
    public void setRas(String r) { ras = r; }
    public void setWarnaBulu(String w) { warnaBulu = w; }

    // Getter
    public String getRas() { return ras; }
    public String getWarnaBulu() { return warnaBulu; }

    @Override
    public void print() {
        super.print();
        System.out.println("Ras                : " + getRas());
        System.out.println("Warna Bulu         : " + getWarnaBulu());
        System.out.println();
    }
}

File: cpp/kucing.cpp
#pragma once
#include <bits/stdc++.h>
#include "hewan.cpp"
using namespace std;

class kucing : public hewan 
{
private:
    string ras;
    string warnaBulu;
public:
    kucing() {}
    kucing(string famili, string habitat, string jenisMakanan, string caraBerkembangbiak, int harapanHidup, makanan food, string ras, string warnaBulu)
        : hewan(famili, habitat, jenisMakanan, caraBerkembangbiak, harapanHidup, food) 
    {
        this->ras = ras;
        this->warnaBulu = warnaBulu;
    }

    // Setter
    void setRas(string r) { ras = r; }
    void setWarnaBulu(string w) { warnaBulu = w; }

    // Getter
    string getRas() { return ras; }
    string getWarnaBulu() { return warnaBulu; }

    void print() override 
    {
        hewan::print();
        cout << "Ras                : " << getRas() << endl;
        cout << "Warna Bulu         : " << getWarnaBulu() << endl << endl;
    }
};
